src/python/WMCore/Cache/WMConfigCache.py

```diff
@@ -432,7 +434,7 @@ def addConfig(self, newConfig, psetHash=None):
 
         """
         # The newConfig parameter is a URL suitable for passing to urlopen.
-        with closing(urllib.urlopen(newConfig)) as f:
+        with closing(urllib.request.urlopen(newConfig)) as f:
```

Comment:
@amaltaro, this line causes the regression https://cmssdt.cern.ch/dmwm-jenkins/view/All/job/DMWM-WMCore-PR-test/10166/testReport/WMCore_t.Cache_t.WMConfigCache_t/testWMConfigCache/testB_addingConfigsAndTweaks/

This is likely a dramatic change in urllib, that apparently does not support file path anymore in py3. more information at https://cms-dmwm-test.docs.cern.ch/py2py3/cheatsheet/urllib_py2/

Some more investigation is required about this matter.

Reply:
The change is far less dramatic than expected. New version of `urllib` require the specifier `file:` to be present in front of a local file path.

Specifier syntax
* Relative path: `file:./relative/path`
* Absolute path: `file://absolute/path`

What can we do here? I updated the urllib_py2 reference, but in short we could do something like

```
## should work in both python 2 and python3
from future import standard_library
standard_library.install_aliases()
import urllib.request
from contextlib import closing
import os

my_path = "./hello_ascii.txt"
my_url = "file://" + urllib.quote(os.path.abspath(my_path)) # escapes nasty characters

with closing(urllib.request.urlopen(my_url)) as f:
    print f.read()
```

This require some manual changes, but they are not going to be wasted time